### ISSUE TITLE
Constant freezing when free memory is low

### ISSUE THREAD

**Issue by mihaivint** [zfsonlinux/zfs#3834]
There is enough cache on the system 4-5 GB but when free memory is low zfs freezes. Both 0.6.5.0 and 0.6.5.1 
System is a virtual machine with 8GB ram 4Gb swap. It's running elasticsearch +rabbitmq on top of zfs. OS is Centos 7 3.10.0-229.11.1.el7.x86_64. Only thing set for zfs is:
```
options zfs zfs_arc_min=536870912
options zfs zfs_arc_max=2147483648
```
It constantly ends up with very high load due to zfs freezes : 
On 0.6.5.0 
```
Sep 23 16:34:26 log kernel: INFO: task kswapd0:47 blocked for more than 120 seconds.
Sep 23 16:34:26 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 23 16:34:26 log kernel: kswapd0         D ffff88023fd13680     0    47      2 0x00000000
Sep 23 16:34:26 log kernel: ffff88023093bbd0 0000000000000046 ffff8802342bf1c0 ffff88023093bfd8
Sep 23 16:34:26 log kernel: ffff88023093bfd8 ffff88023093bfd8 ffff8802342bf1c0 ffff8802306cee00
Sep 23 16:34:26 log kernel: 0000000000000000 00000000006026dc ffff8802306ceec8 00000000006031c0
Sep 23 16:34:26 log kernel: Call Trace:
Sep 23 16:34:26 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 23 16:34:26 log kernel: [<ffffffffa000dd1d>] taskq_wait_outstanding+0x9d/0x110 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 23 16:34:26 log kernel: [<ffffffffa00b7523>] arc_kmem_reap_now+0xf3/0x100 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00bd40f>] arc_shrinker_func+0xaf/0x170 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffff81169095>] shrink_slab+0x165/0x300
Sep 23 16:34:26 log kernel: [<ffffffff811c0381>] ? vmpressure+0x21/0x90
Sep 23 16:34:26 log kernel: [<ffffffff8116cce1>] balance_pgdat+0x4b1/0x5e0
Sep 23 16:34:26 log kernel: [<ffffffff8116cf83>] kswapd+0x173/0x450
Sep 23 16:34:26 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 23 16:34:26 log kernel: [<ffffffff8116ce10>] ? balance_pgdat+0x5e0/0x5e0
Sep 23 16:34:26 log kernel: [<ffffffff8109726f>] kthread+0xcf/0xe0
Sep 23 16:34:26 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 23 16:34:26 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 23 16:34:26 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 23 16:34:26 log kernel: INFO: task arc_prune:129 blocked for more than 120 seconds.
Sep 23 16:34:26 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 23 16:34:26 log kernel: arc_prune       D ffff88023fc13680     0   129      2 0x00000000
Sep 23 16:34:26 log kernel: ffff880035f638b8 0000000000000046 ffff8802304e4fa0 ffff880035f63fd8
Sep 23 16:34:26 log kernel: ffff880035f63fd8 ffff880035f63fd8 ffff8802304e4fa0 ffff8801492eee88
Sep 23 16:34:26 log kernel: ffff8801492eee38 ffff8801492eee90 ffff8801492eee60 0000000000000000
Sep 23 16:34:26 log kernel: Call Trace:
Sep 23 16:34:26 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 23 16:34:26 log kernel: [<ffffffffa00118d5>] cv_wait_common+0x125/0x150 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 23 16:34:26 log kernel: [<ffffffffa0011915>] __cv_wait+0x15/0x20 [spl]
Sep 23 16:34:26 log kernel: [<ffffffffa00c46db>] dbuf_read+0x4ab/0x930 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00df95b>] ? dnode_rele_and_unlock+0x5b/0x90 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00df9c9>] ? dnode_rele+0x39/0x40 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00cde6d>] ? dmu_buf_hold_noread+0xad/0x140 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00cdf50>] dmu_buf_hold+0x50/0x80 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0131dcd>] zap_idx_to_blk+0xcd/0x180 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0132601>] zap_deref_leaf+0x61/0xa0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa013376f>] fzap_remove+0x3f/0xb0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0136f43>] ? zap_name_alloc+0x73/0xd0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0138e3b>] zap_remove_norm+0x17b/0x1e0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0138eb3>] zap_remove+0x13/0x20 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0133ea4>] zap_remove_int+0x54/0x80 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa01426f4>] zfs_rmnode+0x224/0x350 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffff81607ac2>] ? mutex_lock+0x12/0x2f
Sep 23 16:34:26 log kernel: [<ffffffffa0163d88>] zfs_zinactive+0x168/0x180 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa015d4c7>] zfs_inactive+0x67/0x240 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffff81166859>] ? truncate_pagecache+0x59/0x60
Sep 23 16:34:26 log kernel: [<ffffffffa0174ad3>] zpl_evict_inode+0x43/0x60 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffff811e1f87>] evict+0xa7/0x170
Sep 23 16:34:26 log kernel: [<ffffffff811e208e>] dispose_list+0x3e/0x50
Sep 23 16:34:26 log kernel: [<ffffffff811e2f63>] prune_icache_sb+0x163/0x320
Sep 23 16:34:26 log kernel: [<ffffffff811ca0a6>] prune_super+0xd6/0x1a0
Sep 23 16:34:26 log kernel: [<ffffffffa0153192>] zfs_sb_prune+0xa2/0xc0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0175135>] zpl_prune_sb+0x35/0x50 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00b754f>] arc_prune_task+0x1f/0x50 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa000d7ee>] taskq_thread+0x21e/0x420 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff810a9500>] ? wake_up_state+0x20/0x20
Sep 23 16:34:26 log kernel: [<ffffffffa000d5d0>] ? taskq_thread_spawn+0x60/0x60 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff8109726f>] kthread+0xcf/0xe0
Sep 23 16:34:26 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 23 16:34:26 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 23 16:34:26 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 23 16:34:26 log kernel: INFO: task arc_reclaim:130 blocked for more than 120 seconds.
Sep 23 16:34:26 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 23 16:34:26 log kernel: arc_reclaim     D ffff88023fc93680     0   130      2 0x00000000
Sep 23 16:34:26 log kernel: ffff880035f67db8 0000000000000046 ffff8802304e5b00 ffff880035f67fd8
Sep 23 16:34:26 log kernel: ffff880035f67fd8 ffff880035f67fd8 ffff8802304e5b00 ffff8802306cee00
Sep 23 16:34:26 log kernel: 0000000000000000 00000000006026dc ffff8802306ceec8 00000000006031c0
Sep 23 16:34:26 log kernel: Call Trace:
Sep 23 16:34:26 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 23 16:34:26 log kernel: [<ffffffffa000dd1d>] taskq_wait_outstanding+0x9d/0x110 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 23 16:34:26 log kernel: [<ffffffffa00b7523>] arc_kmem_reap_now+0xf3/0x100 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00bd324>] arc_reclaim_thread+0x1f4/0x230 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00bd130>] ? arc_shrink+0xc0/0xc0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa000c931>] thread_generic_wrapper+0x71/0x80 [spl]
Sep 23 16:34:26 log kernel: [<ffffffffa000c8c0>] ? __thread_exit+0x20/0x20 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff8109726f>] kthread+0xcf/0xe0
Sep 23 16:34:26 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 23 16:34:26 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 23 16:34:26 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 23 16:34:26 log kernel: INFO: task z_iput:707 blocked for more than 120 seconds.
Sep 23 16:34:26 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 23 16:34:26 log kernel: z_iput          D ffff88023fc93680     0   707      2 0x00000000
Sep 23 16:34:26 log kernel: ffff8802329f3950 0000000000000046 ffff8802325eb8e0 ffff8802329f3fd8
Sep 23 16:34:26 log kernel: ffff8802329f3fd8 ffff8802329f3fd8 ffff8802325eb8e0 ffff8801492eee88
Sep 23 16:34:26 log kernel: ffff8801492eee38 ffff8801492eee90 ffff8801492eee60 0000000000000000
Sep 23 16:34:26 log kernel: Call Trace:
Sep 23 16:34:26 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 23 16:34:26 log kernel: [<ffffffffa00118d5>] cv_wait_common+0x125/0x150 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 23 16:34:26 log kernel: [<ffffffffa0011915>] __cv_wait+0x15/0x20 [spl]
Sep 23 16:34:26 log kernel: [<ffffffffa00c46db>] dbuf_read+0x4ab/0x930 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00df95b>] ? dnode_rele_and_unlock+0x5b/0x90 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00df9c9>] ? dnode_rele+0x39/0x40 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00cde6d>] ? dmu_buf_hold_noread+0xad/0x140 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00cdf50>] dmu_buf_hold+0x50/0x80 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0131dcd>] zap_idx_to_blk+0xcd/0x180 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0132601>] zap_deref_leaf+0x61/0xa0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00cf0a7>] ? dmu_object_info_from_db+0x37/0x40 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0133385>] fzap_add_cd+0x55/0x140 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa000a9f7>] ? spl_kmem_alloc+0xc7/0x170 [spl]
Sep 23 16:34:26 log kernel: [<ffffffffa01334da>] fzap_add+0x6a/0x80 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0138866>] zap_add+0x126/0x180 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0133e28>] zap_add_int+0x68/0x90 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0100030>] ? gzip_compress+0xa0/0xa0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa014170c>] zfs_unlinked_add+0x3c/0x90 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa01426db>] zfs_rmnode+0x20b/0x350 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0163d88>] zfs_zinactive+0x168/0x180 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa015d4c7>] zfs_inactive+0x67/0x240 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffff81166859>] ? truncate_pagecache+0x59/0x60
Sep 23 16:34:26 log kernel: [<ffffffffa0174ad3>] zpl_evict_inode+0x43/0x60 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffff811e1f87>] evict+0xa7/0x170
Sep 23 16:34:26 log kernel: [<ffffffff811e27c5>] iput+0xf5/0x180
Sep 23 16:34:26 log kernel: [<ffffffffa000d7ee>] taskq_thread+0x21e/0x420 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff810a9500>] ? wake_up_state+0x20/0x20
Sep 23 16:34:26 log kernel: [<ffffffffa000d5d0>] ? taskq_thread_spawn+0x60/0x60 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff8109726f>] kthread+0xcf/0xe0
Sep 23 16:34:26 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 23 16:34:26 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 23 16:34:26 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 23 16:34:26 log kernel: INFO: task txg_quiesce:717 blocked for more than 120 seconds.
Sep 23 16:34:26 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 23 16:34:26 log kernel: txg_quiesce     D ffff88023fd93680     0   717      2 0x00000000
Sep 23 16:34:26 log kernel: ffff88023327bd38 0000000000000046 ffff880232996660 ffff88023327bfd8
Sep 23 16:34:26 log kernel: ffff88023327bfd8 ffff88023327bfd8 ffff880232996660 ffff88022e7840b8
Sep 23 16:34:26 log kernel: ffff88022e784038 ffff88022e7840c0 ffff88022e784060 0000000000000000
Sep 23 16:34:26 log kernel: Call Trace:
Sep 23 16:34:26 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 23 16:34:26 log kernel: [<ffffffffa00118d5>] cv_wait_common+0x125/0x150 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 23 16:34:26 log kernel: [<ffffffffa0011915>] __cv_wait+0x15/0x20 [spl]
Sep 23 16:34:26 log kernel: [<ffffffffa01220db>] txg_quiesce_thread+0x2eb/0x400 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0121df0>] ? txg_init+0x2b0/0x2b0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa000c931>] thread_generic_wrapper+0x71/0x80 [spl]
Sep 23 16:34:26 log kernel: [<ffffffffa000c8c0>] ? __thread_exit+0x20/0x20 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff8109726f>] kthread+0xcf/0xe0
Sep 23 16:34:26 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 23 16:34:26 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 23 16:34:26 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 23 16:34:26 log kernel: INFO: task java:2821 blocked for more than 120 seconds.
Sep 23 16:34:26 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 23 16:34:26 log kernel: java            D ffff88023fc13680     0  2821      1 0x00000080
Sep 23 16:34:26 log kernel: ffff8802318e3920 0000000000000086 ffff88021e4acfa0 ffff8802318e3fd8
Sep 23 16:34:26 log kernel: ffff8802318e3fd8 ffff8802318e3fd8 ffff88021e4acfa0 ffffffffa022d6e0
Sep 23 16:34:26 log kernel: ffffffffa022d7a0 ffffffffa022d6e8 ffffffffa022d7c8 0000000000000000
Sep 23 16:34:26 log kernel: Call Trace:
Sep 23 16:34:26 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 23 16:34:26 log kernel: [<ffffffffa00118d5>] cv_wait_common+0x125/0x150 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 23 16:34:26 log kernel: [<ffffffffa0011915>] __cv_wait+0x15/0x20 [spl]
Sep 23 16:34:26 log kernel: [<ffffffffa00b89c7>] arc_get_data_buf.isra.21+0x3e7/0x3f0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00bbd3f>] arc_buf_alloc+0x10f/0x190 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00bc5a5>] arc_read+0x1c5/0xad0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c2a40>] ? dbuf_rele_and_unlock+0x460/0x460 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c44dd>] dbuf_read+0x2ad/0x930 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00cdf50>] dmu_buf_hold+0x50/0x80 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0137418>] zap_lockdir+0x58/0x900 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0137ebc>] zap_cursor_retrieve+0x1fc/0x320 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa010a8fc>] ? sa_lookup+0x9c/0xc0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffff811da35d>] ? filldir+0x7d/0xf0
Sep 23 16:34:26 log kernel: [<ffffffffa015856e>] zfs_readdir+0x14e/0x4c0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffff811e65ee>] ? mntput_no_expire+0x3e/0x120
Sep 23 16:34:26 log kernel: [<ffffffff811e66f4>] ? mntput+0x24/0x40
Sep 23 16:34:26 log kernel: [<ffffffff811d6736>] ? path_openat+0x166/0x490
Sep 23 16:34:26 log kernel: [<ffffffffa0106b4b>] ? rrm_exit+0x4b/0x90 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa01733f6>] zpl_readdir+0x76/0xc0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffff811da2e0>] ? fillonedir+0xe0/0xe0
Sep 23 16:34:26 log kernel: [<ffffffff811da2e0>] ? fillonedir+0xe0/0xe0
Sep 23 16:34:26 log kernel: [<ffffffff811da1d0>] vfs_readdir+0xb0/0xe0
Sep 23 16:34:26 log kernel: [<ffffffff811da5f5>] SyS_getdents+0x95/0x120
Sep 23 16:34:26 log kernel: [<ffffffff81614289>] system_call_fastpath+0x16/0x1b
Sep 23 16:34:26 log kernel: INFO: task java:2907 blocked for more than 120 seconds.
Sep 23 16:34:26 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 23 16:34:26 log kernel: java            D ffff88023fd93680     0  2907      1 0x00000080
Sep 23 16:34:26 log kernel: ffff88009b3e79a8 0000000000000086 ffff88023426f1c0 ffff88009b3e7fd8
Sep 23 16:34:26 log kernel: ffff88009b3e7fd8 ffff88009b3e7fd8 ffff88023426f1c0 ffffffffa022d6e0
Sep 23 16:34:26 log kernel: ffffffffa022d7a0 ffffffffa022d6e8 ffffffffa022d7c8 0000000000000000
Sep 23 16:34:26 log kernel: Call Trace:
Sep 23 16:34:26 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 23 16:34:26 log kernel: [<ffffffffa00118d5>] cv_wait_common+0x125/0x150 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 23 16:34:26 log kernel: [<ffffffffa0011915>] __cv_wait+0x15/0x20 [spl]
Sep 23 16:34:26 log kernel: [<ffffffffa00b89c7>] arc_get_data_buf.isra.21+0x3e7/0x3f0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00bbd3f>] arc_buf_alloc+0x10f/0x190 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00bc5a5>] arc_read+0x1c5/0xad0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c2a40>] ? dbuf_rele_and_unlock+0x460/0x460 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c44dd>] dbuf_read+0x2ad/0x930 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c4fd8>] __dbuf_hold_impl+0x478/0x500 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c50d3>] dbuf_hold_impl+0x73/0xa0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c540f>] dbuf_hold+0x2f/0x60 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00ccdbd>] dmu_buf_hold_array_by_dnode+0xfd/0x4d0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa000a9f7>] ? spl_kmem_alloc+0xc7/0x170 [spl]
Sep 23 16:34:26 log kernel: [<ffffffffa00cda92>] dmu_read_uio_dnode+0x52/0x100 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00ceb47>] dmu_read_uio_dbuf+0x47/0x60 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0156a0d>] zfs_read+0x13d/0x400 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0173700>] zpl_read_common_iovec.constprop.10+0x80/0xd0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa017385e>] zpl_read+0x7e/0xb0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffff811c66ec>] vfs_read+0x9c/0x170
Sep 23 16:34:26 log kernel: [<ffffffff811c73b2>] SyS_pread64+0x92/0xc0
Sep 23 16:34:26 log kernel: [<ffffffff81614289>] system_call_fastpath+0x16/0x1b
Sep 23 16:34:26 log kernel: INFO: task java:2908 blocked for more than 120 seconds.
Sep 23 16:34:26 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 23 16:34:26 log kernel: java            D ffff88023fc13680     0  2908      1 0x00000080
Sep 23 16:34:26 log kernel: ffff88009b3eba90 0000000000000086 ffff88023426cfa0 ffff88009b3ebfd8
Sep 23 16:34:26 log kernel: ffff88009b3ebfd8 ffff88009b3ebfd8 ffff88023426cfa0 ffffffffa022d6e0
Sep 23 16:34:26 log kernel: ffffffffa022d7a0 ffffffffa022d6e8 ffffffffa022d7c8 0000000000000000
Sep 23 16:34:26 log kernel: Call Trace:
Sep 23 16:34:26 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 23 16:34:26 log kernel: [<ffffffffa00118d5>] cv_wait_common+0x125/0x150 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 23 16:34:26 log kernel: [<ffffffffa0011915>] __cv_wait+0x15/0x20 [spl]
Sep 23 16:34:26 log kernel: [<ffffffffa00b89c7>] arc_get_data_buf.isra.21+0x3e7/0x3f0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00bc715>] arc_read+0x335/0xad0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c40a6>] ? dbuf_create+0x386/0x4e0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c2a40>] ? dbuf_rele_and_unlock+0x460/0x460 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c44dd>] dbuf_read+0x2ad/0x930 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00ccde3>] dmu_buf_hold_array_by_dnode+0x123/0x4d0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00cda92>] dmu_read_uio_dnode+0x52/0x100 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00ceb47>] dmu_read_uio_dbuf+0x47/0x60 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0156a0d>] zfs_read+0x13d/0x400 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0173700>] zpl_read_common_iovec.constprop.10+0x80/0xd0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa017385e>] zpl_read+0x7e/0xb0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffff811c66ec>] vfs_read+0x9c/0x170
Sep 23 16:34:26 log kernel: [<ffffffff811c73b2>] SyS_pread64+0x92/0xc0
Sep 23 16:34:26 log kernel: [<ffffffff81614289>] system_call_fastpath+0x16/0x1b
Sep 23 16:34:26 log kernel: INFO: task java:2909 blocked for more than 120 seconds.
Sep 23 16:34:26 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 23 16:34:26 log kernel: java            D ffff88023fc13680     0  2909      1 0x00000080
Sep 23 16:34:26 log kernel: ffff88009b3ef9a8 0000000000000086 ffff8802308ead80 ffff88009b3effd8
Sep 23 16:34:26 log kernel: ffff88009b3effd8 ffff88009b3effd8 ffff8802308ead80 ffffffffa022d6e0
Sep 23 16:34:26 log kernel: ffffffffa022d7a0 ffffffffa022d6e8 ffffffffa022d7c8 0000000000000000
Sep 23 16:34:26 log kernel: Call Trace:
Sep 23 16:34:26 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 23 16:34:26 log kernel: [<ffffffffa00118d5>] cv_wait_common+0x125/0x150 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 23 16:34:26 log kernel: [<ffffffffa0011915>] __cv_wait+0x15/0x20 [spl]
Sep 23 16:34:26 log kernel: [<ffffffffa00b89c7>] arc_get_data_buf.isra.21+0x3e7/0x3f0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00bbd3f>] arc_buf_alloc+0x10f/0x190 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00bc5a5>] arc_read+0x1c5/0xad0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c2a40>] ? dbuf_rele_and_unlock+0x460/0x460 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c44dd>] dbuf_read+0x2ad/0x930 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c4fd8>] __dbuf_hold_impl+0x478/0x500 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c50d3>] dbuf_hold_impl+0x73/0xa0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c540f>] dbuf_hold+0x2f/0x60 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00ccdbd>] dmu_buf_hold_array_by_dnode+0xfd/0x4d0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa000a9f7>] ? spl_kmem_alloc+0xc7/0x170 [spl]
Sep 23 16:34:26 log kernel: [<ffffffffa00cda92>] dmu_read_uio_dnode+0x52/0x100 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00ceb47>] dmu_read_uio_dbuf+0x47/0x60 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0156a0d>] zfs_read+0x13d/0x400 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0173700>] zpl_read_common_iovec.constprop.10+0x80/0xd0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa017385e>] zpl_read+0x7e/0xb0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffff811c66ec>] vfs_read+0x9c/0x170
Sep 23 16:34:26 log kernel: [<ffffffff811c73b2>] SyS_pread64+0x92/0xc0
Sep 23 16:34:26 log kernel: [<ffffffff81614289>] system_call_fastpath+0x16/0x1b
Sep 23 16:34:26 log kernel: INFO: task java:2910 blocked for more than 120 seconds.
Sep 23 16:34:26 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 23 16:34:26 log kernel: java            D ffff88023fc93680     0  2910      1 0x00000080
Sep 23 16:34:26 log kernel: ffff88009b3f3a90 0000000000000086 ffff8802308e8b60 ffff88009b3f3fd8
Sep 23 16:34:26 log kernel: ffff88009b3f3fd8 ffff88009b3f3fd8 ffff8802308e8b60 ffffffffa022d6e0
Sep 23 16:34:26 log kernel: ffffffffa022d7a0 ffffffffa022d6e8 ffffffffa022d7c8 0000000000000000
Sep 23 16:34:26 log kernel: Call Trace:
Sep 23 16:34:26 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 23 16:34:26 log kernel: [<ffffffffa00118d5>] cv_wait_common+0x125/0x150 [spl]
Sep 23 16:34:26 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 23 16:34:26 log kernel: [<ffffffffa0011915>] __cv_wait+0x15/0x20 [spl]
Sep 23 16:34:26 log kernel: [<ffffffffa00b89c7>] arc_get_data_buf.isra.21+0x3e7/0x3f0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00bc715>] arc_read+0x335/0xad0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c40a6>] ? dbuf_create+0x386/0x4e0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c2a40>] ? dbuf_rele_and_unlock+0x460/0x460 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00c44dd>] dbuf_read+0x2ad/0x930 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00ccde3>] dmu_buf_hold_array_by_dnode+0x123/0x4d0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00cda92>] dmu_read_uio_dnode+0x52/0x100 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa00ceb47>] dmu_read_uio_dbuf+0x47/0x60 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0156a0d>] zfs_read+0x13d/0x400 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa0173700>] zpl_read_common_iovec.constprop.10+0x80/0xd0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffffa017385e>] zpl_read+0x7e/0xb0 [zfs]
Sep 23 16:34:26 log kernel: [<ffffffff811c66ec>] vfs_read+0x9c/0x170
Sep 23 16:34:26 log kernel: [<ffffffff811c73b2>] SyS_pread64+0x92/0xc0
Sep 23 16:34:26 log kernel: [<ffffffff81614289>] system_call_fastpath+0x16/0x1b
```

On 0.6.5.1
```
Sep 24 20:41:11 log kernel: INFO: task kthreadd:2 blocked for more than 120 seconds.
Sep 24 20:41:11 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 24 20:41:11 log kernel: kthreadd        D ffff88023fd13680     0     2      0 0x00000000
Sep 24 20:41:11 log kernel: ffff88023417b9d0 0000000000000046 ffff880234160b60 ffff88023417bfd8
Sep 24 20:41:11 log kernel: ffff88023417bfd8 ffff88023417bfd8 ffff880234160b60 ffff88022f89ee00
Sep 24 20:41:11 log kernel: 0000000000000000 0000000003c57e3e ffff88022f89eec8 0000000003c5b1da
Sep 24 20:41:11 log kernel: Call Trace:
Sep 24 20:41:11 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 24 20:41:11 log kernel: [<ffffffffa000dd1d>] taskq_wait_outstanding+0x9d/0x110 [spl]
Sep 24 20:41:11 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 24 20:41:11 log kernel: [<ffffffffa00b7523>] arc_kmem_reap_now+0xf3/0x100 [zfs]
Sep 24 20:41:11 log kernel: [<ffffffffa00bd40f>] arc_shrinker_func+0xaf/0x170 [zfs]
Sep 24 20:41:11 log kernel: [<ffffffff81169095>] shrink_slab+0x165/0x300
Sep 24 20:41:11 log kernel: [<ffffffff811c0381>] ? vmpressure+0x21/0x90
Sep 24 20:41:11 log kernel: [<ffffffff8116c1e2>] do_try_to_free_pages+0x3c2/0x4e0
Sep 24 20:41:11 log kernel: [<ffffffff8116c3fc>] try_to_free_pages+0xfc/0x180
Sep 24 20:41:11 log kernel: [<ffffffff81160a7d>] __alloc_pages_nodemask+0x7fd/0xb90
Sep 24 20:41:11 log kernel: [<ffffffff810b1c00>] ? update_curr+0x50/0x150
Sep 24 20:41:11 log kernel: [<ffffffff8106bee4>] copy_process.part.25+0x134/0x14d0
Sep 24 20:41:11 log kernel: [<ffffffff810abe95>] ? sched_clock_cpu+0xb5/0x100
Sep 24 20:41:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:41:11 log kernel: [<ffffffff8106d43c>] do_fork+0xbc/0x350
Sep 24 20:41:11 log kernel: [<ffffffff8106d6f6>] kernel_thread+0x26/0x30
Sep 24 20:41:11 log kernel: [<ffffffff81097d72>] kthreadd+0x2b2/0x2f0
Sep 24 20:41:11 log kernel: [<ffffffff81097ac0>] ? kthread_create_on_cpu+0x60/0x60
Sep 24 20:41:11 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 24 20:41:11 log kernel: [<ffffffff81097ac0>] ? kthread_create_on_cpu+0x60/0x60
Sep 24 20:41:11 log kernel: INFO: task kswapd0:47 blocked for more than 120 seconds.
Sep 24 20:41:11 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 24 20:41:11 log kernel: kswapd0         D ffff88023fd93680     0    47      2 0x00000000
Sep 24 20:41:11 log kernel: ffff88023093bbd0 0000000000000046 ffff880230930000 ffff88023093bfd8
Sep 24 20:41:11 log kernel: ffff88023093bfd8 ffff88023093bfd8 ffff880230930000 ffff88022f89ee00
Sep 24 20:41:11 log kernel: 0000000000000000 0000000003c57e3e ffff88022f89eec8 0000000003c5b1da
Sep 24 20:41:11 log kernel: Call Trace:
Sep 24 20:41:11 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 24 20:41:11 log kernel: [<ffffffffa000dd1d>] taskq_wait_outstanding+0x9d/0x110 [spl]
Sep 24 20:41:11 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 24 20:41:11 log kernel: [<ffffffffa00b7523>] arc_kmem_reap_now+0xf3/0x100 [zfs]
Sep 24 20:41:11 log kernel: [<ffffffffa00bd40f>] arc_shrinker_func+0xaf/0x170 [zfs]
Sep 24 20:41:11 log kernel: [<ffffffff81169095>] shrink_slab+0x165/0x300
Sep 24 20:41:11 log kernel: [<ffffffff811c0381>] ? vmpressure+0x21/0x90
Sep 24 20:41:11 log kernel: [<ffffffff8116cce1>] balance_pgdat+0x4b1/0x5e0
Sep 24 20:41:11 log kernel: [<ffffffff8116cf83>] kswapd+0x173/0x450
Sep 24 20:41:11 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 24 20:41:11 log kernel: [<ffffffff8116ce10>] ? balance_pgdat+0x5e0/0x5e0
Sep 24 20:41:11 log kernel: [<ffffffff8109726f>] kthread+0xcf/0xe0
Sep 24 20:41:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:41:11 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 24 20:41:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:41:11 log kernel: INFO: task spl_dynamic_tas:114 blocked for more than 120 seconds.
Sep 24 20:41:11 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 24 20:41:11 log kernel: spl_dynamic_tas D ffff88023fc93680     0   114      2 0x00000000
Sep 24 20:41:11 log kernel: ffff8802304f7b70 0000000000000046 ffff8802304971c0 ffff8802304f7fd8
Sep 24 20:41:11 log kernel: ffff8802304f7fd8 ffff8802304f7fd8 ffff8802304971c0 ffff8802304f7cd8
Sep 24 20:41:11 log kernel: ffff8802304f7ce0 7fffffffffffffff ffff8802304971c0 ffff8802304f9800
Sep 24 20:41:11 log kernel: Call Trace:
Sep 24 20:41:11 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 24 20:41:11 log kernel: [<ffffffff81607679>] schedule_timeout+0x209/0x2d0
Sep 24 20:41:11 log kernel: [<ffffffff810ae7a2>] ? select_task_rq_fair+0x552/0x6f0
Sep 24 20:41:11 log kernel: [<ffffffff8104a3c3>] ? x2apic_send_IPI_mask+0x13/0x20
Sep 24 20:41:11 log kernel: [<ffffffff81609c26>] wait_for_completion+0x116/0x170
Sep 24 20:41:11 log kernel: [<ffffffff810a9500>] ? wake_up_state+0x20/0x20
Sep 24 20:41:11 log kernel: [<ffffffffa000d5d0>] ? taskq_thread_spawn+0x60/0x60 [spl]
Sep 24 20:41:11 log kernel: [<ffffffff81097108>] kthread_create_on_node+0xa8/0x140
Sep 24 20:41:11 log kernel: [<ffffffff812df19b>] ? string.isra.6+0x3b/0xf0
Sep 24 20:41:11 log kernel: [<ffffffffa000d5d0>] ? taskq_thread_spawn+0x60/0x60 [spl]
Sep 24 20:41:11 log kernel: [<ffffffffa000d5d0>] ? taskq_thread_spawn+0x60/0x60 [spl]
Sep 24 20:41:11 log kernel: [<ffffffffa000c9dc>] spl_kthread_create+0x9c/0xf0 [spl]
Sep 24 20:41:11 log kernel: [<ffffffffa000dffb>] taskq_thread_create+0x6b/0x110 [spl]
Sep 24 20:41:11 log kernel: [<ffffffffa000e0b2>] taskq_thread_spawn_task+0x12/0x30 [spl]
Sep 24 20:41:11 log kernel: [<ffffffffa000d7ee>] taskq_thread+0x21e/0x420 [spl]
Sep 24 20:41:11 log kernel: [<ffffffff810a9500>] ? wake_up_state+0x20/0x20
Sep 24 20:41:11 log kernel: [<ffffffffa000d5d0>] ? taskq_thread_spawn+0x60/0x60 [spl]
Sep 24 20:41:11 log kernel: [<ffffffff8109726f>] kthread+0xcf/0xe0
Sep 24 20:41:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:41:11 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 24 20:41:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:41:11 log kernel: INFO: task arc_reclaim:130 blocked for more than 120 seconds.
Sep 24 20:41:11 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 24 20:41:11 log kernel: arc_reclaim     D ffff88023fc13680     0   130      2 0x00000000
Sep 24 20:41:11 log kernel: ffff8800bb803db8 0000000000000046 ffff88022f8e16c0 ffff8800bb803fd8
Sep 24 20:41:11 log kernel: ffff8800bb803fd8 ffff8800bb803fd8 ffff88022f8e16c0 ffff88022f89ee00
Sep 24 20:41:11 log kernel: 0000000000000000 0000000003c57e3e ffff88022f89eec8 0000000003c5b1da
Sep 24 20:41:11 log kernel: Call Trace:
Sep 24 20:41:11 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 24 20:41:11 log kernel: [<ffffffffa000dd1d>] taskq_wait_outstanding+0x9d/0x110 [spl]
Sep 24 20:41:11 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 24 20:41:11 log kernel: [<ffffffffa00b7523>] arc_kmem_reap_now+0xf3/0x100 [zfs]
Sep 24 20:41:11 log kernel: [<ffffffffa00bd324>] arc_reclaim_thread+0x1f4/0x230 [zfs]
Sep 24 20:41:11 log kernel: [<ffffffffa00bd130>] ? arc_shrink+0xc0/0xc0 [zfs]
Sep 24 20:41:11 log kernel: [<ffffffffa000c931>] thread_generic_wrapper+0x71/0x80 [spl]
Sep 24 20:41:11 log kernel: [<ffffffffa000c8c0>] ? __thread_exit+0x20/0x20 [spl]
Sep 24 20:41:11 log kernel: [<ffffffff8109726f>] kthread+0xcf/0xe0
Sep 24 20:41:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:41:11 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 24 20:41:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:43:11 log kernel: INFO: task kthreadd:2 blocked for more than 120 seconds.
Sep 24 20:43:11 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 24 20:43:11 log kernel: kthreadd        D ffff88023fd13680     0     2      0 0x00000000
Sep 24 20:43:11 log kernel: ffff88023417b9d0 0000000000000046 ffff880234160b60 ffff88023417bfd8
Sep 24 20:43:11 log kernel: ffff88023417bfd8 ffff88023417bfd8 ffff880234160b60 ffff88022f89ee00
Sep 24 20:43:11 log kernel: 0000000000000000 0000000003c57e3e ffff88022f89eec8 0000000003c5b1da
Sep 24 20:43:11 log kernel: Call Trace:
Sep 24 20:43:11 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 24 20:43:11 log kernel: [<ffffffffa000dd1d>] taskq_wait_outstanding+0x9d/0x110 [spl]
Sep 24 20:43:11 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 24 20:43:11 log kernel: [<ffffffffa00b7523>] arc_kmem_reap_now+0xf3/0x100 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa00bd40f>] arc_shrinker_func+0xaf/0x170 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffff81169095>] shrink_slab+0x165/0x300
Sep 24 20:43:11 log kernel: [<ffffffff811c0381>] ? vmpressure+0x21/0x90
Sep 24 20:43:11 log kernel: [<ffffffff8116c1e2>] do_try_to_free_pages+0x3c2/0x4e0
Sep 24 20:43:11 log kernel: [<ffffffff8116c3fc>] try_to_free_pages+0xfc/0x180
Sep 24 20:43:11 log kernel: [<ffffffff81160a7d>] __alloc_pages_nodemask+0x7fd/0xb90
Sep 24 20:43:11 log kernel: [<ffffffff810b1c00>] ? update_curr+0x50/0x150
Sep 24 20:43:11 log kernel: [<ffffffff8106bee4>] copy_process.part.25+0x134/0x14d0
Sep 24 20:43:11 log kernel: [<ffffffff810abe95>] ? sched_clock_cpu+0xb5/0x100
Sep 24 20:43:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:43:11 log kernel: [<ffffffff8106d43c>] do_fork+0xbc/0x350
Sep 24 20:43:11 log kernel: [<ffffffff8106d6f6>] kernel_thread+0x26/0x30
Sep 24 20:43:11 log kernel: [<ffffffff81097d72>] kthreadd+0x2b2/0x2f0
Sep 24 20:43:11 log kernel: [<ffffffff81097ac0>] ? kthread_create_on_cpu+0x60/0x60
Sep 24 20:43:11 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 24 20:43:11 log kernel: [<ffffffff81097ac0>] ? kthread_create_on_cpu+0x60/0x60
Sep 24 20:43:11 log kernel: INFO: task kswapd0:47 blocked for more than 120 seconds.
Sep 24 20:43:11 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 24 20:43:11 log kernel: kswapd0         D ffff88023fd93680     0    47      2 0x00000000
Sep 24 20:43:11 log kernel: ffff88023093bbd0 0000000000000046 ffff880230930000 ffff88023093bfd8
Sep 24 20:43:11 log kernel: ffff88023093bfd8 ffff88023093bfd8 ffff880230930000 ffff88022f89ee00
Sep 24 20:43:11 log kernel: 0000000000000000 0000000003c57e3e ffff88022f89eec8 0000000003c5b1da
Sep 24 20:43:11 log kernel: Call Trace:
Sep 24 20:43:11 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 24 20:43:11 log kernel: [<ffffffffa000dd1d>] taskq_wait_outstanding+0x9d/0x110 [spl]
Sep 24 20:43:11 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 24 20:43:11 log kernel: [<ffffffffa00b7523>] arc_kmem_reap_now+0xf3/0x100 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa00bd40f>] arc_shrinker_func+0xaf/0x170 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffff81169095>] shrink_slab+0x165/0x300
Sep 24 20:43:11 log kernel: [<ffffffff811c0381>] ? vmpressure+0x21/0x90
Sep 24 20:43:11 log kernel: [<ffffffff8116cce1>] balance_pgdat+0x4b1/0x5e0
Sep 24 20:43:11 log kernel: [<ffffffff8116cf83>] kswapd+0x173/0x450
Sep 24 20:43:11 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 24 20:43:11 log kernel: [<ffffffff8116ce10>] ? balance_pgdat+0x5e0/0x5e0
Sep 24 20:43:11 log kernel: [<ffffffff8109726f>] kthread+0xcf/0xe0
Sep 24 20:43:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:43:11 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 24 20:43:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:43:11 log kernel: INFO: task khugepaged:49 blocked for more than 120 seconds.
Sep 24 20:43:11 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 24 20:43:11 log kernel: khugepaged      D ffff88023fd13680     0    49      2 0x00000000
Sep 24 20:43:11 log kernel: ffff88023098b4e8 0000000000000046 ffff8802309316c0 ffff88023098bfd8
Sep 24 20:43:11 log kernel: ffff88023098bfd8 ffff88023098bfd8 ffff8802309316c0 ffffffffa022d6e0
Sep 24 20:43:11 log kernel: ffffffffa022d7a0 ffffffffa022d6e8 ffffffffa022d7c8 0000000000000000
Sep 24 20:43:11 log kernel: Call Trace:
Sep 24 20:43:11 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 24 20:43:11 log kernel: [<ffffffffa00118d5>] cv_wait_common+0x125/0x150 [spl]
Sep 24 20:43:11 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 24 20:43:11 log kernel: [<ffffffffa0011915>] __cv_wait+0x15/0x20 [spl]
Sep 24 20:43:11 log kernel: [<ffffffffa00b89c7>] arc_get_data_buf.isra.21+0x3e7/0x3f0 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa00bbd3f>] arc_buf_alloc+0x10f/0x190 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa00c4e5a>] __dbuf_hold_impl+0x2fa/0x500 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa00c50d3>] dbuf_hold_impl+0x73/0xa0 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa00c540f>] dbuf_hold+0x2f/0x60 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa00cde47>] dmu_buf_hold_noread+0x87/0x140 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa00cdf2f>] dmu_buf_hold+0x2f/0x80 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa013233c>] zap_get_leaf_byblk+0x5c/0x2c0 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa00c2bae>] ? dmu_buf_rele+0xe/0x10 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa0131e03>] ? zap_idx_to_blk+0x103/0x180 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa013261a>] zap_deref_leaf+0x7a/0xa0 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa013376f>] fzap_remove+0x3f/0xb0 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa0136f43>] ? zap_name_alloc+0x73/0xd0 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa0138e3b>] zap_remove_norm+0x17b/0x1e0 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa0138eb3>] zap_remove+0x13/0x20 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa0133ea4>] zap_remove_int+0x54/0x80 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa01426f4>] zfs_rmnode+0x224/0x350 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffff81607ac2>] ? mutex_lock+0x12/0x2f
Sep 24 20:43:11 log kernel: [<ffffffffa0163d88>] zfs_zinactive+0x168/0x180 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa015d4c7>] zfs_inactive+0x67/0x240 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffff81166859>] ? truncate_pagecache+0x59/0x60
Sep 24 20:43:11 log kernel: [<ffffffffa0174ad3>] zpl_evict_inode+0x43/0x60 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffff811e1f87>] evict+0xa7/0x170
Sep 24 20:43:11 log kernel: [<ffffffff811e208e>] dispose_list+0x3e/0x50
Sep 24 20:43:11 log kernel: [<ffffffff811e2f63>] prune_icache_sb+0x163/0x320
Sep 24 20:43:11 log kernel: [<ffffffff811ca0a6>] prune_super+0xd6/0x1a0
Sep 24 20:43:11 log kernel: [<ffffffff81169095>] shrink_slab+0x165/0x300
Sep 24 20:43:11 log kernel: [<ffffffff8116c1e2>] do_try_to_free_pages+0x3c2/0x4e0
Sep 24 20:43:11 log kernel: [<ffffffff8116c3fc>] try_to_free_pages+0xfc/0x180
Sep 24 20:43:11 log kernel: [<ffffffff81160a7d>] __alloc_pages_nodemask+0x7fd/0xb90
Sep 24 20:43:11 log kernel: [<ffffffff811b4129>] khugepaged_scan_mm_slot+0x3c9/0xc60
Sep 24 20:43:11 log kernel: [<ffffffff811b4c17>] khugepaged+0x257/0x480
Sep 24 20:43:11 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 24 20:43:11 log kernel: [<ffffffff811b49c0>] ? khugepaged_scan_mm_slot+0xc60/0xc60
Sep 24 20:43:11 log kernel: [<ffffffff8109726f>] kthread+0xcf/0xe0
Sep 24 20:43:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:43:11 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 24 20:43:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:43:11 log kernel: INFO: task spl_dynamic_tas:114 blocked for more than 120 seconds.
Sep 24 20:43:11 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 24 20:43:11 log kernel: spl_dynamic_tas D ffff88023fc93680     0   114      2 0x00000000
Sep 24 20:43:11 log kernel: ffff8802304f7b70 0000000000000046 ffff8802304971c0 ffff8802304f7fd8
Sep 24 20:43:11 log kernel: ffff8802304f7fd8 ffff8802304f7fd8 ffff8802304971c0 ffff8802304f7cd8
Sep 24 20:43:11 log kernel: ffff8802304f7ce0 7fffffffffffffff ffff8802304971c0 ffff8802304f9800
Sep 24 20:43:11 log kernel: Call Trace:
Sep 24 20:43:11 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 24 20:43:11 log kernel: [<ffffffff81607679>] schedule_timeout+0x209/0x2d0
Sep 24 20:43:11 log kernel: [<ffffffff810ae7a2>] ? select_task_rq_fair+0x552/0x6f0
Sep 24 20:43:11 log kernel: [<ffffffff8104a3c3>] ? x2apic_send_IPI_mask+0x13/0x20
Sep 24 20:43:11 log kernel: [<ffffffff81609c26>] wait_for_completion+0x116/0x170
Sep 24 20:43:11 log kernel: [<ffffffff810a9500>] ? wake_up_state+0x20/0x20
Sep 24 20:43:11 log kernel: [<ffffffffa000d5d0>] ? taskq_thread_spawn+0x60/0x60 [spl]
Sep 24 20:43:11 log kernel: [<ffffffff81097108>] kthread_create_on_node+0xa8/0x140
Sep 24 20:43:11 log kernel: [<ffffffff812df19b>] ? string.isra.6+0x3b/0xf0
Sep 24 20:43:11 log kernel: [<ffffffffa000d5d0>] ? taskq_thread_spawn+0x60/0x60 [spl]
Sep 24 20:43:11 log kernel: [<ffffffffa000d5d0>] ? taskq_thread_spawn+0x60/0x60 [spl]
Sep 24 20:43:11 log kernel: [<ffffffffa000c9dc>] spl_kthread_create+0x9c/0xf0 [spl]
Sep 24 20:43:11 log kernel: [<ffffffffa000dffb>] taskq_thread_create+0x6b/0x110 [spl]
Sep 24 20:43:11 log kernel: [<ffffffffa000e0b2>] taskq_thread_spawn_task+0x12/0x30 [spl]
Sep 24 20:43:11 log kernel: [<ffffffffa000d7ee>] taskq_thread+0x21e/0x420 [spl]
Sep 24 20:43:11 log kernel: [<ffffffff810a9500>] ? wake_up_state+0x20/0x20
Sep 24 20:43:11 log kernel: [<ffffffffa000d5d0>] ? taskq_thread_spawn+0x60/0x60 [spl]
Sep 24 20:43:11 log kernel: [<ffffffff8109726f>] kthread+0xcf/0xe0
Sep 24 20:43:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:43:11 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 24 20:43:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:43:11 log kernel: INFO: task arc_reclaim:130 blocked for more than 120 seconds.
Sep 24 20:43:11 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 24 20:43:11 log kernel: arc_reclaim     D ffff88023fc13680     0   130      2 0x00000000
Sep 24 20:43:11 log kernel: ffff8800bb803db8 0000000000000046 ffff88022f8e16c0 ffff8800bb803fd8
Sep 24 20:43:11 log kernel: ffff8800bb803fd8 ffff8800bb803fd8 ffff88022f8e16c0 ffff88022f89ee00
Sep 24 20:43:11 log kernel: 0000000000000000 0000000003c57e3e ffff88022f89eec8 0000000003c5b1da
Sep 24 20:43:11 log kernel: Call Trace:
Sep 24 20:43:11 log kernel: [<ffffffff81609729>] schedule+0x29/0x70
Sep 24 20:43:11 log kernel: [<ffffffffa000dd1d>] taskq_wait_outstanding+0x9d/0x110 [spl]
Sep 24 20:43:11 log kernel: [<ffffffff81098230>] ? wake_up_bit+0x30/0x30
Sep 24 20:43:11 log kernel: [<ffffffffa00b7523>] arc_kmem_reap_now+0xf3/0x100 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa00bd324>] arc_reclaim_thread+0x1f4/0x230 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa00bd130>] ? arc_shrink+0xc0/0xc0 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa000c931>] thread_generic_wrapper+0x71/0x80 [spl]
Sep 24 20:43:11 log kernel: [<ffffffffa000c8c0>] ? __thread_exit+0x20/0x20 [spl]
Sep 24 20:43:11 log kernel: [<ffffffff8109726f>] kthread+0xcf/0xe0
Sep 24 20:43:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:43:11 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 24 20:43:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:43:11 log kernel: INFO: task z_wr_iss:678 blocked for more than 120 seconds.
Sep 24 20:43:11 log kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Sep 24 20:43:11 log kernel: z_wr_iss        D ffff88023fc13680     0   678      2 0x00000000
Sep 24 20:43:11 log kernel: ffff8800bb30bc80 0000000000000046 ffff88022f75b8e0 ffff8800bb30bfd8
Sep 24 20:43:11 log kernel: ffff8800bb30bfd8 ffff8800bb30bfd8 ffff88022f75b8e0 ffff88008d1079c8
Sep 24 20:43:11 log kernel: ffff88008d1079cc ffff88022f75b8e0 00000000ffffffff ffff88008d1079d0
Sep 24 20:43:11 log kernel: Call Trace:
Sep 24 20:43:11 log kernel: [<ffffffff8160a919>] schedule_preempt_disabled+0x29/0x70
Sep 24 20:43:11 log kernel: [<ffffffff81608665>] __mutex_lock_slowpath+0xc5/0x1c0
Sep 24 20:43:11 log kernel: [<ffffffff81607acf>] mutex_lock+0x1f/0x2f
Sep 24 20:43:11 log kernel: [<ffffffffa00c1f55>] dbuf_write_ready+0x85/0x210 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa00b7798>] arc_write_ready+0x28/0xc0 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa016e91c>] zio_ready+0x6c/0x470 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa000cc82>] ? taskq_member+0x62/0x70 [spl]
Sep 24 20:43:11 log kernel: [<ffffffffa016ae98>] zio_execute+0xc8/0x180 [zfs]
Sep 24 20:43:11 log kernel: [<ffffffffa000d7ee>] taskq_thread+0x21e/0x420 [spl]
Sep 24 20:43:11 log kernel: [<ffffffff810a9500>] ? wake_up_state+0x20/0x20
Sep 24 20:43:11 log kernel: [<ffffffffa000d5d0>] ? taskq_thread_spawn+0x60/0x60 [spl]
Sep 24 20:43:11 log kernel: [<ffffffff8109726f>] kthread+0xcf/0xe0
Sep 24 20:43:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
Sep 24 20:43:11 log kernel: [<ffffffff816141d8>] ret_from_fork+0x58/0x90
Sep 24 20:43:11 log kernel: [<ffffffff810971a0>] ? kthread_create_on_node+0x140/0x140
```

**Comment by behlendorf** [comment]
This is a duplicate of #3808 and there is fix in #3834 which will appear in the next point release.

**Comment by odoucet** [comment]
@behlendorf you linked this issue, not the fix that will appear in the next point release ;)

**Comment by behlendorf** [comment]
@odoucet whoops!  Fixed, sorry I was off by a digit.  Pull request #3826 contains the fix.

**Comment by behlendorf** [comment]
Closing as duplicate of #3808.

**Comment by rfehren** [comment]
It seems this problem is still not completely fixed. With kernel 3.12.x and 0.65.2 My nodes still hang reproducably with swap on a zvol. Very simple test case: Run something creating permanent disk I/O like e.g. a fio job and then start a Linpack job in parallel that is active on all cores and consumes all RAM. After some minutes you will see the familiar hung tasks messages and a little later a total dead-lock.  
```[ 8292.590906] INFO: task kswapd0:145 blocked for more than 120 seconds.
[ 8292.597367]       Tainted: P           O 3.12.48-ql-generic-9.1-61 #1
[ 8292.603825] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[ 8292.611812] INFO: task kswapd1:146 blocked for more than 120 seconds.
[ 8292.618264]       Tainted: P           O 3.12.48-ql-generic-9.1-61 #1
[ 8292.624719] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[ 8292.632710] INFO: task rpcbind:1855 blocked for more than 120 seconds.
[ 8292.639247]       Tainted: P           O 3.12.48-ql-generic-9.1-61 #1
[ 8292.645701] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[ 8292.653650] INFO: task arc_reclaim:2899 blocked for more than 120 seconds.
[ 8292.660537]       Tainted: P           O 3.12.48-ql-generic-9.1-61 #1
[ 8292.666992] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[ 8292.675546] INFO: task txg_sync:3753 blocked for more than 120 seconds.
[ 8292.682456]       Tainted: P           O 3.12.48-ql-generic-9.1-61 #1
[ 8292.689184] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.```
